daemon/errors.go

```diff
@@ -14,16 +14,12 @@ func (d *Daemon) imageNotExistToErrcode(err error) error {
 			e := fmt.Errorf("No such image: %s", dne.RefOrID)
 			return errors.NewRequestNotFoundError(e)
 		}
-		tag := reference.DefaultTag
-		ref, err := reference.ParseNamed(dne.RefOrID)
+		ref, err := reference.ParseNormalizedNamed(dne.RefOrID)
```

Comment:
It wasn't possible here before, but is now because references can also contain bare digests. It could be a followup, but I think it's a reasonable thing to do as part of making this change.